src/loaders/OmeZarrLoader.ts

```diff
@@ -310,22 +395,24 @@ class OMEZarrLoader extends ThreadableVolumeLoader {
   }
 
   /** Reads a list of chunk keys requested by a `loadVolumeData` call and sets up appropriate prefetch requests. */
-  private beginPrefetch(keys: string[], scaleLevel: NumericZarrArray): void {
-    const numDims = getDimensionCount(this.axesTCZYX);
-
+  private beginPrefetch(keys: ZarrChunkFetchInfo[], scaleLevels: NumericZarrArray[]): void {
     // Convert keys to arrays of coords
-    const chunkCoords = keys.map((key) => {
+    const chunkCoords = keys.map(({ sourceIdx, key }) => {
+      const numDims = getDimensionCount(this.sources[sourceIdx].axesTCZYX);
       const coordsInDimensionOrder = key
         .trim()
         .split("/")
         .slice(-numDims)
         .filter((s) => s !== "")
         .map((s) => parseInt(s, 10));
-      return this.orderByTCZYX(coordsInDimensionOrder, 0);
+      const sourceCoords = this.orderByTCZYX(coordsInDimensionOrder, 0, sourceIdx);
+      sourceCoords[1] += this.sources[sourceIdx].channelOffset;
```

Comment:
a comment would be nice to reassure me that we are incrementing the channel index for a good downstream reason because it no longer is a valid "chunk" coordinate (not sure how it gets used)